src/components/client/UploadImage/context/UploadImageContext/UploadImageContext.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import {
+  ChangeEvent,
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { toast } from 'react-toastify';
+
+import { UploadImageContextProps, UploadImageProviderProps } from './UploadImageContext.type';
+
+export const UploadImageContext = createContext<UploadImageContextProps>({
+  images: [],
+  handleUploadImage: () => {},
+  handleRemoveImage: () => {},
+});
+
+export const UploadImageProvider = ({
+  limit = 1,
+  onChange,
+  children,
+}: UploadImageProviderProps) => {
+  const [images, setImages] = useState<File[]>([]);
+  const onChangeRef = useRef(onChange);
+
+  const handleUploadImage = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const selectedImages = event.target.files;
+
+      if (!selectedImages) {
+        return;
+      }
+
+      if (selectedImages.length > limit) {
+        return toast.warn(`최대 ${limit}만큼 업로드가 가능합니다.`);
+      }
+
+      const imagesArray = Array.from(selectedImages);
+
+      onChangeRef.current(imagesArray);
+      setImages(imagesArray);
+    },
+    [limit],
+  );
+
+  const handleRemoveImage = useCallback(
+    (selectedImage: File) => {
+      const newImagesArray = images.filter((image) => image !== selectedImage);
+
+      onChangeRef.current(newImagesArray);
+      setImages(newImagesArray);
+    },
+    [images],
+  );
+
+  const value = useMemo(
+    () => ({
+      images,
+      handleUploadImage,
+      handleRemoveImage,
+    }),
+    [images, handleUploadImage, handleRemoveImage],
+  );
+
+  useEffect(() => {
+    onChangeRef.current = onChange;
+  }, [onChange]);
+
+  return <UploadImageContext.Provider value={value}>{children}</UploadImageContext.Provider>;
+};
```

Comment:
[ P2 ] 
해당 Context를 사용하는 Component 내부를 보니 대부분
별도로 

```
const { handleRemoveImage } = useContext(UploadImageContext);
```
`useContext`와 `UploadImageContext`를 항상 불러서 사용하고 있더라구요

물론 성능상으로는 문제없지만 코드수를 줄일 수 있을것같아서 Context 파일내부에서 

```suggestion
};

const useUploadImageContext = () => {
  const contextValue = useContext(UploadImageContext);
  return contextValue;
};
```

와 같이 간단한 hook을 내보내주어 `useContext` 와 `UploadImageContext`를 지속해서 호출 하는 부분을 `useUploadImageContext` 하나만 호출하면 되도록 변경해보면 어떨까요?

Reply:
오 좋은 것 같습니다 

아니면 혹시 

export const context = useContext(UploadImageContext); 방식은 어떤가요

Reply:
정상적인 출력이 가능하다면 괜찮을것 같은데 안된다면 위에 제시한 코드로 해보는건어떨까요!